.github/workflows/testcontainers-integration-tests.yml

```diff
@@ -0,0 +1,29 @@
+name: Test Containers Integration Tests
+
+on:
+  push:
+    branches: [main]
+  pull_request:
+    branches: [main]
+  repository_dispatch:
+    types: [ok-to-test]
+
+jobs:
+  test-build-and-run-testcontainers:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Set up Go
+        uses: actions/setup-go@v4
+        with:
+          go-version: '1.23'
```

Comment:
Maybe there is a better way to get the go-version we are using in the project, but here is an approach we are using in `swagger-validation.yml`

https://github.com/bacalhau-project/bacalhau/blob/85cbe5b3fa662191c19b2599ee32232a89c6753f/.github/workflows/swagger-validation.yml#L20-L33

Also, I've previously noticed Post Setup Go takes minutes with `setup-go@v4` and I ended up updating to `v5` to solve it. It doesn't seem it is taking long in your case though

Reply:
@wdbaruni thanks for the pointer. Resolved.